mangaki/mangaki/tests/test_anidb.py

```diff
@@ -47,5 +52,19 @@ def test_anidb_get(self):
             content_type='application/xml'
         )
         anime = self.anidb.get_or_update_work(11606)
-        self.assertNotEqual(anime.title, '')
-        self.assertNotEqual(anime.worktitle_set.count(), 0)
+
+        self.assertEqual(anime.title, 'Sangatsu no Lion')
+        self.assertEqual(anime.nb_episodes, 22)
+        self.assertEqual(anime.studio.title, 'Shaft')
+
+        author_ids = [q.artist_id for q in Staff.objects.filter(work_id=anime.pk, role_id=3)]
+        author_names = [Artist.objects.get(pk=at_id).name for at_id in author_ids]
+        self.assertEqual(author_names, ['Umino Chika'])
```

Comment:
## Another way to get `author_names`

Finally, getting the author names of a given work could be done using the related fields.

First, `Staff` has a relationship of type Many to One with `Work` (it's marked by the `ForeignKey('work')` which is on `Staff` model).

Now, as a result, `Work` get a special "invisible" field called `staff_set` which returns a queryset of `Staff` which all respects the condition `Staff.work = Work considered`.

Example: `Work.objects.get(id=1).staff_set.all()` would return the list of staff of the work with `id=1`.

Knowing this, you could fetch all staff members using `anime.staff_set` and then partition the list by `role` (a sort of `group_by`).

Finally, you could access to the name using `staff.artist.name`.

The question is: which one is faster? which one should we use? Let me know your opinion on this.